libs/sdk-core/src/breez_services.rs

```diff
@@ -1896,26 +1896,49 @@ impl Receiver for PaymentReceiver {
     }
 }
 
-/// Convenience method to look up LSP info based on current LSP ID
-async fn get_lsp(persister: Arc<SqliteStorage>, lsp: Arc<dyn LspAPI>) -> Result<LspInformation> {
+/// Convenience method to look up LSP info based on current LSP ID and
+/// fallback to the first LSP in the listed LSPs if the current LSP is not available.
+async fn get_lsp(
```

Comment:
It seems that if we change the selected lsp in this method it may  change it even if the user didn't set it directly.
For example:
1. User set explicit lsp id in the config
2. Once sdk is connected the lsp changes without the user being aware of it. Perhaps the expected behavior in this case is to actually get an error "lsp not found"

I am proposing here an alternative:
1. On our default config remove the constant lsp_id (leave None). Let the user set it if desired.
2. Make the change to select dynamic lsp if not already selected at the `connect_lsp_peer` function that is invoked automatically on every sync and on startup.

Reply:
Makes sense because it keeps the logic in the connect_lsp_peer fn. Have issues with it at the moment as the node_state is not persisted at that point, which is used to fetch the lsp list